dartagnan/src/main/java/com/dat3m/dartagnan/wmm/Wmm.java

```diff
@@ -80,6 +83,14 @@ public void addRecursiveGroup(Set<RecursiveRelation> recursiveGroup){
         recursiveGroups.add(new RecursiveGroup(id, recursiveGroup));
     }
 
+	/**
```

Comment:
This is my viewpoint on this method.  I emphasize on the change of state in `Wmm` instances and the objects it owns.  The doc misses the parameter `ctx`, because I did not find any uses of it in the call hierarchy.